From d29b84fe5076fa0af9606a8462fc8cf2bc3001ce Mon Sep 17 00:00:00 2001
From: tedyu <yuzhihong@gmail.com>
Date: Sat, 20 Aug 2016 17:18:27 -0700
Subject: [PATCH 356/362] HBASE-16409 Row key for bad row should be properly
 delimited in VerifyReplication

HBASE-16409 Addendum fixes the case where delimiter is not specified by user

(cherry picked from commit d260108e39eb41853fa5856885db82236cf3dbca)
(cherry picked from commit c78a5fd6500f18c99b203b7d85684d2e5e6d346e)

Change-Id: I90dfdde32437bf3161f650e19b0feefc514b3929
Reason:    Supportability
Author:    tedyu <yuzhihong@gmail.com>
Ref:       CDH-54606
---
 .../mapreduce/replication/VerifyReplication.java   |   13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/replication/VerifyReplication.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/replication/VerifyReplication.java
index d8cc6c5..82958a7 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/replication/VerifyReplication.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/replication/VerifyReplication.java
@@ -74,6 +74,7 @@ public class VerifyReplication extends Configured implements Tool {
   static int versions = -1;
   static String tableName = null;
   static String families = null;
+  static String delimiter = "";
   static String peerId = null;
 
   /**
@@ -173,7 +174,8 @@ public class VerifyReplication extends Configured implements Tool {
     private void logFailRowAndIncreaseCounter(Context context, Counters counter, Result row) {
       context.getCounter(counter).increment(1);
       context.getCounter(Counters.BADROWS).increment(1);
-      LOG.error(counter.toString() + ", rowkey=" + Bytes.toStringBinary(row.getRow()));
+      LOG.error(counter.toString() + ", rowkey=" + delimiter + Bytes.toStringBinary(row.getRow()) +
+          delimiter);
     }
 
     @Override
@@ -341,6 +343,12 @@ public class VerifyReplication extends Configured implements Tool {
           printUsage("Invalid argument '" + cmd + "'");
         }
 
+        final String delimiterArgKey = "--delimiter=";
+        if (cmd.startsWith(delimiterArgKey)) {
+          delimiter = cmd.substring(delimiterArgKey.length());
+          continue;
+        }
+
         if (i == args.length-2) {
           peerId = cmd;
         }
@@ -365,7 +373,7 @@ public class VerifyReplication extends Configured implements Tool {
       System.err.println("ERROR: " + errorMsg);
     }
     System.err.println("Usage: verifyrep [--starttime=X]" +
-        " [--endtime=Y] [--families=A] <peerid> <tablename>");
+        " [--endtime=Y] [--families=A] [--delimiter=] <peerid> <tablename>");
     System.err.println();
     System.err.println("Options:");
     System.err.println(" starttime    beginning of the time range");
@@ -373,6 +381,7 @@ public class VerifyReplication extends Configured implements Tool {
     System.err.println(" endtime      end of the time range");
     System.err.println(" versions     number of cell versions to verify");
     System.err.println(" families     comma-separated list of families to copy");
+    System.err.println(" delimiter    the delimiter used in display around rowkey");
     System.err.println();
     System.err.println("Args:");
     System.err.println(" peerid       Id of the peer used for verification, must match the one given for replication");
-- 
1.7.9.5

